### PR TITLE
Bats tests for charsets and collations

### PR DESCRIPTION
**Pull request by timsehn**
This is a first pass and includes skipped tests for the bugs: #4172, #4170, #4168

### CODE REVIEW

**Comment by timsehn** [comment]
This PR https://github.com/dolthub/go-mysql-server/pull/1215 has fixes for a couple of the skipped tests. Will update once it is merged.

**Comment by timsehn** [comment]
@Hydrocharged this is ready for your review

**Comment by timsehn** [comment]
@Hydrocharged let me know if you want me to add a skipped test for the shell printing issue we have been discussing. I switched the existing tests to counts to get around it but maybe it's better to have a test that uses shell regexes?

File: integration-tests/bats/sql-charsets-collations.bats [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+    server_query "test" 1 dolt "" "use test; SELECT @@character_set_database" ";@@SESSION.collation_database\nlatin1_swedish_ci"
+}
+
+@test "sql-charsets-collations: define and use a colation and charset" {
+    dolt sql -q "create table german1 (c char(10) CHARACTER SET latin1 COLLATE latin1_german1_ci)"
+    dolt sql -q "insert into german1 values ('Bar'), ('Bär')"
+    run dolt sql -q "SELECT * FROM german1 WHERE c = 'Bär'"
+    [ $status -eq 0 ]
+    [[ $output =~ 'Bar' ]] || false
+    # Hard to match a latin1 characeter in bash
```
Comment (Hydrocharged) [inline integration-tests/bats/sql-charsets-collations.bats:54]:
```suggestion
    # Hard to match a latin1 character in bash
```

File: integration-tests/bats/sql-charsets-collations.bats [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+    [ ${#lines[@]} -eq 6 ]
+
+    dolt sql -q	"create table german2 (c char(10) CHARACTER SET latin1 COLLATE latin1_german2_ci)"
+    dolt sql -q	"insert into german2 values ('Bar'), ('Bär')"
+    run dolt sql -q "SELECT * FROM german2 WHERE c = 'Bär'"
+    [ $status -eq 0 ]
+    [[ ! $output =~ 'Bar' ]] || false
+    # Hard to match a latin1 characeter in bash
```
Comment (Hydrocharged) [inline integration-tests/bats/sql-charsets-collations.bats:62]:
```suggestion
    # Hard to match a latin1 character in bash
```

File: integration-tests/bats/sql-charsets-collations.bats [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+@test "sql-charsets-collations: define charset and collation on a database" {
+    start_sql_server
+
+    server_query "" 1 dolt "" "CREATE DATABASE test CHARACTER SET latin1 COLLATE latin1_swedish_ci;"
+    skip "Defining charsets and collations on a database not supported"
+    server_query "test" 1 dolt "" "use test; SELECT @@character_set_database" ";@@SESSION.character_set_database\nlatin1"
+    server_query "test" 1 dolt "" "use test; SELECT @@character_set_database" ";@@SESSION.collation_database\nlatin1_swedish_ci"
+}
+
+@test "sql-charsets-collations: define and use a colation and charset" {
```
Comment (Hydrocharged) [inline integration-tests/bats/sql-charsets-collations.bats:48]:
```suggestion
@test "sql-charsets-collations: define and use a collation and charset" {
```

**Review by Hydrocharged — APPROVED** [review]
LGTM, just some small typos

**Comment by Hydrocharged** [comment]
Yeah we should add a skipped test for the printing issue. Verify the system variable’s value as well.

**Pull request merged**